frontend/packages/operator-lifecycle-manager/src/dev-catalog.ts

```diff
@@ -4,6 +4,22 @@ import { ClusterServiceVersionKind } from './types';
 import { referenceForProvidedAPI, providedAPIsFor } from './components';
 import * as operatorLogo from './operator.svg';
 
+const isInternal = (crd: { name: string }): boolean => {
+  const internalOpListString = _.get(
+    crd,
+    ['csv', 'metadata', 'annotations', 'operators.operatorframework.io/internal-objects'],
+    '[]',
+  );
+  try {
+    const internalOpList = internalOpListString.slice(1, -1).split(','); // JSON.parse fails or returns improper structure
```

Comment:
Should this be a valid JSON string? If so, we might want to look a little bit deeper into why `JSON.parse` fails.

Reply:
I think we need to assume that the annotation is a valid JSON string and fail if it isn't. Otherwise, we leave this open to all kinds of edge cases where the string is malformed.

Reply:
Agreed. If there is a value, `try` / `catch` the `JSON.parse` call. If it fails, move on.

Reply:
Using JSON.parse in try and console.error in catch